source/state_representation/include/state_representation/robot/JointState.hpp

```diff
@@ -42,232 +42,312 @@ class JointState : public State {
   static JointState Zero(const std::string& robot_name, const std::vector<std::string>& joint_names);
 
   /**
-   * @brief Constructor for the random JointState
-   * @param robot_name the name of the associated robot
-   * @param nb_joints the number of joints for initialization
+   * @brief Constructor for a random JointState.
+   * @param robot_name The name of the associated robot
+   * @param nb_joints The number of joints for initialization
    * @return JointState with random values in all attributes
    */
   static JointState Random(const std::string& robot_name, unsigned int nb_joints);
 
   /**
-   * @brief Constructor for the random JointState
-   * @param robot_name the name of the associated robot
-   * @param joint_names list of joint names
+   * @brief Constructor for a random JointState.
+   * @param robot_name The name of the associated robot
+   * @param joint_names List of joint names
    * @return JointState with random values in all attributes
    */
   static JointState Random(const std::string& robot_name, const std::vector<std::string>& joint_names);
 
   /**
-   * @brief Swap the values of the two JointState
+   * @brief Swap the values of the two JointState.
    * @param state1 JointState to be swapped with 2
    * @param state2 JointState to be swapped with 1
    */
   friend void swap(JointState& state1, JointState& state2);
 
   /**
-   * @brief Copy assignment operator that have to be defined to the custom assignment operator
-   * @param state the state with value to assign
-   * @return reference to the current state with new values
+   * @brief Copy assignment operator that have to be defined to the custom assignment operator.
+   * @param state The state with value to assign
+   * @return Reference to the current state with new values
    */
   JointState& operator=(const JointState& state);
 
   /**
-   * @brief Getter of the size from the attributes
+   * @brief Getter of the size from the attributes.
+   * @return The size of the joint state
    */
   unsigned int get_size() const;
 
   /**
-   * @brief Getter of the names attribute
+   * @brief Getter of the names attribute.
+   * @return The vector of strings containing the joint names
    */
   const std::vector<std::string>& get_names() const;
 
   /**
-   * @brief Setter of the names attribute from the number of joints
+   * @brief Setter of the names attribute from the number of joints.
+   * @param nb_joints The number of joints of the joint state
    */
   void set_names(unsigned int nb_joints);
 
   /**
-   * @brief Setter of the names attribute
+   * @brief Setter of the names attribute.
+   * @param names The vector of strings containing the joint names
    */
   void set_names(const std::vector<std::string>& names);
 
   /**
-   * @brief Getter of the positions attribute
+   * @brief Get joint index by the name of the joint, if it exists.
+   * @param joint_name The name of the desired joint
+   * @return The index of the joint, if it exists
```

Comment:
This would actually be nice to add on each scalar getter too, just for explicitness. You can use `@copydetails JointState::get_joint_index()` and it will automatically fill in.